evm-contracts/bsc-accounting-contract/src/AccountingContract.sol

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.19;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract AccountingContract is Ownable {
+    constructor(
+        address initialOwner,
+        address initialWorker
+    ) Ownable(initialOwner) {
+        require(initialWorker != address(0), "Worker can't be zero address");
+        require(initialOwner != address(0), "Owner can't be zero address");
+        admin = initialOwner;
+        worker = initialWorker;
+    }
+
+    address public admin;
+    address public worker;
+    mapping(address => uint256) public nonces;
+
+    struct Payout {
+        uint256 amount;
+        bool paid;
+    }
+
+    mapping(address => mapping(uint256 => Payout)) public payouts;
+
+    event AdminUpdated(address indexed oldAdmin, address indexed newAdmin);
+    event WorkerUpdated(address indexed oldWorker, address indexed newWorker);
+    event FundsDeposited(address indexed from, uint256 amount);
+    event FundsWithdrawn(address indexed to, uint256 amount);
+    event PayoutRequestCreated(
+        address indexed beneficiary,
+        uint256 amount,
+        uint256 nonce
+    );
+
+    modifier onlyAdmin() {
+        require(msg.sender == admin, "Unauthorized: not admin");
+        _;
+    }
+
+    modifier onlyWorker() {
+        require(msg.sender == worker, "Unauthorized: not worker");
+        _;
+    }
+
+    function setAdmin(address newAdmin) external onlyOwner {
```

Comment:
I kind of feel `Owner` and `Admin` are somewhat overlapped, is it better to keep only one?

Reply:
Yeah, I felt so, So i updated the contract to use `AccessControls` instead of `Ownable`, I'm still keeping the storage item for worker public key so that when setting new worker, we can also revoke the role for the current worker role holder, to ensure only one public key will always have the `Worker` role.